packages/codegen/src/generateEntityCodegenFile.ts

```diff
@@ -646,14 +646,16 @@ function generateGraphQLFilterFields(meta: EntityDbMetadata): Code[] {
   const pgEnums = meta.pgEnums.map(({ fieldName, enumType }) => {
     return code`${fieldName}?: ${EnumGraphQLFilter}<${enumType}>;`;
   });
-  const m2o = meta.manyToOnes.map(({ fieldName, otherEntity }) => {
-    return code`${fieldName}?: ${EntityGraphQLFilter}<${otherEntity.type}, ${otherEntity.idType}, ${GraphQLFilterOf}<${otherEntity.type}>>;`;
+  const m2o = meta.manyToOnes.map(({ fieldName, otherEntity, notNull }) => {
+    return code`${fieldName}?: ${EntityGraphQLFilter}<${otherEntity.type}, ${otherEntity.idType}, ${GraphQLFilterOf}<${
+      otherEntity.type
+    }>, ${nullOrNever(notNull)}>;`;
   });
   const o2o = meta.oneToOnes.map(({ fieldName, otherEntity }) => {
-    return code`${fieldName}?: ${EntityGraphQLFilter}<${otherEntity.type}, ${otherEntity.idType}, ${GraphQLFilterOf}<${otherEntity.type}>>;`;
+    return code`${fieldName}?: ${EntityGraphQLFilter}<${otherEntity.type}, ${otherEntity.idType}, ${GraphQLFilterOf}<${otherEntity.type}>, null | undefined>;`;
   });
   const polys = meta.polymorphics.map(({ fieldName, fieldType }) => {
-    return code`${fieldName}?: ${EntityGraphQLFilter}<${fieldType}, ${IdOf}<${fieldType}>, never>;`;
+    return code`${fieldName}?: ${EntityGraphQLFilter}<${fieldType}, ${IdOf}<${fieldType}>, never, null | undefined>;`;
```

Comment:
Thought: I think polys can be either nullable or not-nullable, so we could probably do a conditional here as well?

Reply:
I was following the example from `EntityFilter` above, which assumed `null | undefined`
![image](https://user-images.githubusercontent.com/24765506/174667687-08a410cc-f9b1-4f60-ab6b-f46d12a25a55.png)

I could try adding that in, but feel like I would like to add some more test cases around both. Looks like we currently only have a `notNull` polymorphic relationship.

Reply:
Cool, yeah sgtm to punt. If anything this is the more permissive version.

Reply:
Looked into that a bit. I think it warrants a separate follow up PR, since this work as-is will unblock some work we need to do.

Reply:
Cool! Feel free to merge/release when you're ready